dace/subsets.py

```diff
@@ -107,14 +165,20 @@ def covers_precise(self, other):
 
         # If self does not cover other with a bounding box union, return false.
         symbolic_positive = Config.get('optimizer', 'symbolic_positive')
-        try:
-            bounding_box_cover = bounding_box_cover_exact(
-                self, other) if symbolic_positive else bounding_box_symbolic_positive(self, other)
-            if not bounding_box_cover:
-                return False
-        except TypeError:
+        if symbolic_positive and (not bounding_box_cover_exact(self, other)):
             return False
+        elif not bounding_box_symbolic_positive(self, other):
+            return False
+
+        # TODO: The original implementation always called `nng()` on the argument before passing
+        #   it into the simplify function. This looks like an error, calling it should depend
+        #   on the value of `symbolic_positive`.
+        simplify = lambda expr: symbolic.simplify_ext(nng(expr))
```

Comment:
Have you tested what happens to the test cases when that is changed to what seems like a fix?